terraform/modules/shared/main.tf

```diff
@@ -472,8 +472,52 @@ resource "azurerm_kubernetes_cluster" "cluster" {
   }
 
   identity {
-    type         = "UserAssigned"
-    identity_ids = [azurerm_user_assigned_identity.pipeline_runner.id]
+    type = "SystemAssigned"
+  }
+
+}
+
+data "azurerm_kubernetes_cluster" "credentials" {
+  name                = azurerm_kubernetes_cluster.cluster.name
+  resource_group_name = var.resource_group_name
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.azurerm_kubernetes_cluster.credentials.kube_config.0.host
```

Comment:
There are a number of ways to authenticate with a Helm provider in Terraform, but this seemed like the most common way to do it for Azure.

Reply:
This looks fine to me if it works.